### ISSUE TITLE
Build failure on stackage for 0.4.1 (dependency upgrade?)

### ISSUE THREAD

**Issue by mihaimaruseac** [centromere/pipes-cacophony#4]
Full log

```haskell
[1 of 1] Compiling Pipes.Noise      ( src/Pipes/Noise.hs, dist/build/Pipes/Noise.o )

src/Pipes/Noise.hs:51:20: error:
    • Couldn't match expected type ‘ScrubbedBytes’
                  with actual type ‘NoiseState c d h’
    • In the first argument of ‘readMessage’, namely ‘ns’
      In the expression: readMessage ns msg
      In a stmt of a 'do' block:
        case readMessage ns msg of {
          Left e -> return . Left $ e
          Right (pt, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield pt;
                    .... } }
    • Relevant bindings include
        ns :: NoiseState c d h (bound at src/Pipes/Noise.hs:50:3)
        nsmv :: MVar (NoiseState c d h) (bound at src/Pipes/Noise.hs:47:13)
        inboundPipe :: MVar (NoiseState c d h)
                       -> InboundNoisePipe m (Either SomeException ())
          (bound at src/Pipes/Noise.hs:47:1)

src/Pipes/Noise.hs:51:23: error:
    • Couldn't match expected type ‘NoiseState c1 d1 h1’
                  with actual type ‘ByteString’
    • In the second argument of ‘readMessage’, namely ‘msg’
      In the expression: readMessage ns msg
      In a stmt of a 'do' block:
        case readMessage ns msg of {
          Left e -> return . Left $ e
          Right (pt, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield pt;
                    .... } }

src/Pipes/Noise.hs:52:5: error:
    • Couldn't match expected type ‘NoiseResult c1 d1 h1’
                  with actual type ‘Either SomeException t2’
    • In the pattern: Left e
      In a case alternative: Left e -> return . Left $ e
      In a stmt of a 'do' block:
        case readMessage ns msg of {
          Left e -> return . Left $ e
          Right (pt, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield pt;
                    .... } }

src/Pipes/Noise.hs:53:5: error:
    • Couldn't match expected type ‘NoiseResult c1 d1 h1’
                  with actual type ‘Either t3 (ScrubbedBytes, NoiseState c d h)’
    • In the pattern: Right (pt, ns')
      In a case alternative:
          Right (pt, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield pt;
                    inboundPipe nsmv }
      In a stmt of a 'do' block:
        case readMessage ns msg of {
          Left e -> return . Left $ e
          Right (pt, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield pt;
                    .... } }
    • Relevant bindings include
        ns :: NoiseState c d h (bound at src/Pipes/Noise.hs:50:3)
        nsmv :: MVar (NoiseState c d h) (bound at src/Pipes/Noise.hs:47:13)
        inboundPipe :: MVar (NoiseState c d h)
                       -> InboundNoisePipe m (Either SomeException ())
          (bound at src/Pipes/Noise.hs:47:1)

src/Pipes/Noise.hs:65:21: error:
    • Couldn't match expected type ‘ScrubbedBytes’
                  with actual type ‘NoiseState c d h’
    • In the first argument of ‘writeMessage’, namely ‘ns’
      In the expression: writeMessage ns msg
      In a stmt of a 'do' block:
        case writeMessage ns msg of {
          Left e -> return . Left $ e
          Right (ct, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield ct;
                    .... } }
    • Relevant bindings include
        ns :: NoiseState c d h (bound at src/Pipes/Noise.hs:64:3)
        nsmv :: MVar (NoiseState c d h) (bound at src/Pipes/Noise.hs:61:14)
        outboundPipe :: MVar (NoiseState c d h)
                        -> OutboundNoisePipe m (Either SomeException ())
          (bound at src/Pipes/Noise.hs:61:1)

src/Pipes/Noise.hs:65:24: error:
    • Couldn't match expected type ‘NoiseState c0 d0 h0’
                  with actual type ‘ScrubbedBytes’
    • In the second argument of ‘writeMessage’, namely ‘msg’
      In the expression: writeMessage ns msg
      In a stmt of a 'do' block:
        case writeMessage ns msg of {
          Left e -> return . Left $ e
          Right (ct, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield ct;
                    .... } }

src/Pipes/Noise.hs:66:5: error:
    • Couldn't match expected type ‘NoiseResult c0 d0 h0’
                  with actual type ‘Either SomeException t0’
    • In the pattern: Left e
      In a case alternative: Left e -> return . Left $ e
      In a stmt of a 'do' block:
        case writeMessage ns msg of {
          Left e -> return . Left $ e
          Right (ct, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield ct;
                    .... } }

src/Pipes/Noise.hs:67:5: error:
    • Couldn't match expected type ‘NoiseResult c0 d0 h0’
                  with actual type ‘Either t1 (ByteString, NoiseState c d h)’
    • In the pattern: Right (ct, ns')
      In a case alternative:
          Right (ct, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield ct;
                    outboundPipe nsmv }
      In a stmt of a 'do' block:
        case writeMessage ns msg of {
          Left e -> return . Left $ e
          Right (ct, ns')
            -> do { liftIO . putMVar nsmv $ ns';
                    yield ct;
                    .... } }
    • Relevant bindings include
        ns :: NoiseState c d h (bound at src/Pipes/Noise.hs:64:3)
        nsmv :: MVar (NoiseState c d h) (bound at src/Pipes/Noise.hs:61:14)
        outboundPipe :: MVar (NoiseState c d h)
                        -> OutboundNoisePipe m (Either SomeException ())
          (bound at src/Pipes/Noise.hs:61:1)
```

**Comment by centromere** [comment]
I've just released 0.5.0. Does this fix the issue?

**Comment by bergmark** [comment]
Looks like it's fixed, thanks!
